### ISSUE TITLE
jinja[spacing] conflict with yaml[line-length]

### ISSUE THREAD

**Issue by sathieu** [ansible/ansible-lint#2522]
##### Summary

With the following `roles/test/tasks/main.yml`:

```yaml
- name: Test
  ansible.builtin.debug:
    msg: "Hello: world"
  vars:
    sealed_raw_input: "{{ {
          cluster_namespace + '/' + cluster_name + '/username': hostvars[control_plane].cluster_api_vsphere_username,
          cluster_namespace + '/' + cluster_name + '/password': hostvars[control_plane].cluster_api_vsphere_password,
      } }}"
```

I get:

```python
$ ansible-lint 
WARNING  Listing 1 violation(s) that are fatal
jinja: Jinja2 spacing could be improved: {{ { cluster_namespace + '/' + cluster_name + '/username': hostvars[control_plane].cluster_api_vsphere_username, cluster_namespace + '/' + cluster_name + '/password': hostvars[control_plane].cluster_api_vsphere_password, } }} -> {{ {
    cluster_namespace + '/' + cluster_name + '/username': hostvars[control_plane].cluster_api_vsphere_username,
    cluster_namespace + '/' + cluster_name + '/password': hostvars[control_plane].cluster_api_vsphere_password,
} }} (jinja[spacing])
roles/test/tasks/main.yml:1 Jinja2 template rewrite recommendation: `{{ {
    cluster_namespace + '/' + cluster_name + '/username': hostvars.cluster_api_vsphere_username,
    cluster_namespace + '/' + cluster_name + '/password': hostvars.cluster_api_vsphere_password,
} }}`.


Finished with 0 failure(s), 1 warning(s) on 2 files.
```

Running `ansible-lint --write`  makes the code less readable:

```diff
diff --git a/roles/test/tasks/main.yml b/roles/test/tasks/main.yml
index b8d293d..a218bfe 100644
--- a/roles/test/tasks/main.yml
+++ b/roles/test/tasks/main.yml
@@ -1,8 +1,7 @@
+---
 - name: Test
   ansible.builtin.debug:
     msg: "Hello: world"
   vars:
-    sealed_raw_input: "{{ {
-          cluster_namespace + '/' + cluster_name + '/username': hostvars[control_plane].cluster_api_vsphere_username,
-          cluster_namespace + '/' + cluster_name + '/password': hostvars[control_plane].cluster_api_vsphere_password,
-      } }}"
+    sealed_raw_input: "{{ { cluster_namespace + '/' + cluster_name + '/username': hostvars[control_plane].cluster_api_vsphere_username, cluster_namespace + '/' +\
+      \ cluster_name + '/password': hostvars[control_plane].cluster_api_vsphere_password, } }}"
```

And also add a new error:

```python
 ansible-lint 
WARNING  Listing 2 violation(s) that are fatal
jinja: Jinja2 spacing could be improved: {{ { cluster_namespace + '/' + cluster_name + '/username': hostvars[control_plane].cluster_api_vsphere_username, cluster_namespace + '/' + cluster_name + '/password': hostvars[control_plane].cluster_api_vsphere_password, } }} -> {{ {
    cluster_namespace + '/' + cluster_name + '/username': hostvars[control_plane].cluster_api_vsphere_username,
    cluster_namespace + '/' + cluster_name + '/password': hostvars[control_plane].cluster_api_vsphere_password,
} }} (jinja[spacing])
roles/test/tasks/main.yml:2 Jinja2 template rewrite recommendation: `{{ {
    cluster_namespace + '/' + cluster_name + '/username': hostvars.cluster_api_vsphere_username,
    cluster_namespace + '/' + cluster_name + '/password': hostvars.cluster_api_vsphere_password,
} }}`.

yaml: line too long (162 > 160 characters) (yaml[line-length])
roles/test/tasks/main.yml:6

You can skip specific rules or tags by adding them to your configuration file:
# .config/ansible-lint.yml
warn_list:  # or 'skip_list' to silence them completely
  - yaml[line-length]  # Violations reported by yamllint.

Finished with 1 failure(s), 1 warning(s) on 2 files.
```

##### Issue Type

- Bug Report

##### Ansible and Ansible Lint details

```console
$ ansible --version
ansible [core 2.13.0]
  config file = None
  configured module search path = ['/home/mathieu/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/mathieu/git/gitlab.com/kubitus-project/kubitus-installer/venv/lib/python3.9/site-packages/ansible
  ansible collection location = /home/mathieu/.ansible/collections:/usr/share/ansible/collections
  executable location = /home/mathieu/git/gitlab.com/kubitus-project/kubitus-installer/venv/bin/ansible
  python version = 3.9.2 (default, Feb 28 2021, 17:03:44) [GCC 10.2.1 20210110]
  jinja version = 3.1.2
  libyaml = True
$ ansible-lint --version
ansible-lint 6.7.0 using ansible 2.13.0
```

- ansible installation method: pip (in virtualenv)
- ansible-lint installation method: pip (in virtualenv)

**Comment by ssbarnea** [comment]
Sorry but there are no current plans to add re-wrap jinja. While @cognifloyd worked towards allowing this, we have to say that for now, we rely on humans to do the wrapping, as only they can decide where it makes more sense to split the line.

Here is a YAML hint to cut few columns:

```
var_name: >
  value...
```

Basically technically doable, but the outcomes almost always sub-par.

**Comment by sathieu** [comment]
Thanks @ssbarnea I'll try this way.
